napari/layers/labels/labels.py

```diff
@@ -743,7 +743,11 @@ def mode(self):
         In ERASE mode the cursor functions similarly to PAINT mode, but to
         paint with background label, which effectively removes the label.
         """
-        return str(self._mode)
+        return Layer.mode.fget(self)
+
+    @mode.setter
+    def mode(self, mode):
+        Layer.mode.fset(self, mode)
```

Comment:
While the original implementation was confusing, I think this is even more confusing and adds another layer of obscurity to everything. I would revert back to using the original `@getter` override, but you can keep the `fget` line with a comment explaining that it's supposed to act like that for inheritance purposes and is just being re-written to change the docstring.

Since it is ultimately just to modify the docstring, I would actually almost rather have something like the following:
```python
mode = property(Layer.mode.fget, Layer.mode.fset, doc="...")
```

Reply:
I went with setting the doc manually after the definition of `Layer`. I think that makes it clear that only the doc is being changed, not any functionality.

Reply:
In the end it was easiest to keep a setter/getter, and add a comment.